docker/reference/reference.go

```diff
@@ -102,13 +107,13 @@ func WithTag(name Named, tag string) (NamedTagged, error) {
 
 // WithDigest combines the name from "name" and the digest from "digest" to form
 // a reference incorporating both the name and the digest.
-func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
-	r, err := distreference.WithDigest(name, digest)
-	if err != nil {
-		return nil, err
-	}
-	return &canonicalRef{namedRef{r}}, nil
-}
+//func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
```

Comment:
Yeah, apparently this is by design. Filed https://github.com/docker/distribution/issues/2130 to make `docker/distribution` aware of the issue.

Could you split this into two commits (commit 1: drop our `WithDigest`; commit 2: move from `docker/distribution/digest` to `opencontainers/go-digest`) so that, if this gets resolved, we can cleanly revert commit 1?